### PR TITLE
fix: improve payment by stripe

### PR DESCRIPTION
**Pull request by pranavpandey1998official**
1)corrected the keyboard type for input

2)made the whole button clickable in case of the month, year and card type selector

3)added card No. validator and automatic card type selector

feat: #1071 
ezgif com-video-to-gif

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt [hunk 1]
```diff
@@ -217,6 +195,54 @@ class AttendeeFragment : Fragment() {
...
+            }
+
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+            }
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                val visaPattern = Pattern.compile("^4[0-9]{0,15}$")
+                val masterCardPattern = Pattern.compile("^(5[1-5]|222[1-9]|22[3-9][0-9]|2[3-6]" +
+                    "[0-9]{2}|27[01][0-9]|2720)[0-9]{0,15}$")
+                val americanExpressPattern = Pattern.compile("^3[47][0-9]{0,15}$")
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: Stripe most probably has pre defined functions to check the above. Can you confirm? if yes use them

**Comment by pranavpandey1998official** [comment]
They have their own cardInputwidget which handles all these operation https://stripe.com/docs/mobile/android .

File: app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt (hunk 1 above)
Comment (pranavpandey1998official) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: They have their own cardInputwidget which handles all these operation https://stripe.com/docs/mobile/android .

- That would mean changing the whole input section .. should I use it ???

Comment (nikit19) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: If that is the case then it is fine

**Comment by iamareebjamal** [comment]
Please add wip: in title until the work is complete

File: app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt (hunk 1 above)
Comment (pranavpandey1998official) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: Can you elaborate is current commit is fine or I should use cardInputWidget

Comment (pranavpandey1998official) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: @nikit19 please reply

Comment (nikit19) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: It is fine

**Comment by pranavpandey1998official** [comment]
@iamareebjamal, please review

**Comment by iamareebjamal** [comment]
Fix coflicts

**Comment by pranavpandey1998official** [comment]
@iamareebjamal done

File: app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt [hunk 2]
```diff
@@ -216,6 +219,54 @@ class AttendeeFragment : Fragment() {
...
+                        cardBrand = attendeeViewModel.cardType[3]
+                        selectCardText.text = cardBrand
+                        rootView.cardSelector.isVisible = false
+                        rootView.cardNumber.error = null
+                    }
+                    else -> {
+                        cardBrand = attendeeViewModel.cardType[0]
+                        selectCardText.text = cardBrand
+                        rootView.cardSelector.isVisible = true
+                        rootView.cardNumber.error = "Invalid card number"
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: Move logic to a util and remove duplication

File: app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt [hunk 3]
```diff
@@ -217,6 +219,33 @@ class AttendeeFragment : Fragment() {
...
+                if (card == -1) {
+                    rootView.cardSelector.setSelection(0, true)
+                    rootView.cardSelector.isVisible = true
+                    rootView.cardNumber.error = "Invalid card number"
+                    return
+                }
+
+                rootView.cardSelector.setSelection(card, true)
+                rootView.cardSelector.isVisible = false
+                rootView.cardNumber.error = null
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: Make a function of these 3 calls

File: app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt [hunk 4]
```diff
@@ -131,4 +132,17 @@ object Utils {
             navigation.animation = AnimationUtils.loadAnimation(context, R.anim.slide_down)
         }
     }
+
+    fun creditCardVerify(s: String): Int {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: Name properly and return enum

Comment (pranavpandey1998official) [reply app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: done

**Comment by iamareebjamal** [comment]
@pranavpandey1998official Updates?

**Comment by pranavpandey1998official** [comment]
@iamareebjamal I have introduced an enum as you said and removed all unnecessary changes. what next?

**Comment by iamareebjamal** [comment]
Go to File changes and see the requested changes

**Comment by iamareebjamal** [comment]
There are still unrelated changes. Please revert them.

**Comment by pranavpandey1998official** [comment]
@iamareebjamal the indentation changes are done through gradlew spotlessApply , if i revert them gradlew spotlessCheck is failing and rest are changes are related to commit please specify if you find something else

**Pull request merged**